hack/rp-dev/rp_funcs.sh

```diff
@@ -0,0 +1,197 @@
+    mdsd_image_tag=$(get_digest_tag "MdsdImage")
+    mdsd_repo=$(echo $mdsd_image_tag | sed -n 's|.*/\([^/]*\/[^:]*\):.*|\1|p')
+    import_geneva_image $mdm_repo $mdm_image_tag
+    import_geneva_image $mdm_repo $mdm_image_tag
+    echo "Success step 6d ✈️ 📦 - Import MDM and MDSD to ACR"
+
+    make publish-image-aro-multistage
+    echo "Success step 6e ✈️ 📦 - Build, push and import to ARO image to ACR"
+
+    fluentbit_image_tag=$(get_digest_tag "FluentbitImage")
+    copy_digest_tag $PULL_SECRET "arointsvc" $DST_ACR_NAME $fluentbit_image_tag
+    echo "Success step 6f ✈️ 📦 - Copy Fluenbit image to ACR"
+  else
+    echo "⏩📋 Skip mirroring repos to ACR $DST_ACR_NAME" 
+  fi
+  echo -e "Success step 6 ✅ - Mirror repos to ACR $DST_ACR_NAME\n"
+}
+
+prepare_RP_deployment() {
+  export PARENT_DOMAIN_NAME=osadev.cloud
+  export GLOBAL_RESOURCEGROUP=${AZURE_PREFIX}-global
+
+  for DOMAIN_NAME in ${AZURE_PREFIX}-clusters.$PARENT_DOMAIN_NAME ${AZURE_PREFIX}-rp.$PARENT_DOMAIN_NAME; do
+    CHILD_DOMAIN_PREFIX="$(cut -d. -f1 <<<$DOMAIN_NAME)"
+    echo "########## Creating NS record to DNS Zone $CHILD_DOMAIN_PREFIX ##########"
+    az network dns record-set ns create \
+      --resource-group "$PARENT_DOMAIN_RESOURCEGROUP" \
+      --zone "$PARENT_DOMAIN_NAME" \
+      --name "$CHILD_DOMAIN_PREFIX" >/dev/null
+    for ns in $(az network dns zone show \
+      --resource-group "$GLOBAL_RESOURCEGROUP" \
+      --name "$DOMAIN_NAME" \
+      --query nameServers -o tsv); do
+      az network dns record-set ns add-record \
+      --resource-group "$PARENT_DOMAIN_RESOURCEGROUP" \
+      --zone "$PARENT_DOMAIN_NAME" \
+      --record-set-name "$CHILD_DOMAIN_PREFIX" \
+      --nsdname "$ns" >/dev/null
+    done
+  done
+  echo "Success step 7a ✅ - Update DNS Child Domains"
+
+  check_and_import_certificates
+  echo "Success step 7b ✅ - Update certificates in keyvault"
+
+  gitCommit=$(git rev-parse --short=7 HEAD)
+  az vmss delete --resource-group ${AZURE_PREFIX}-aro-$LOCATION --name rp-vmss-$gitCommit --force-deletion
+  az vmss delete --resource-group ${AZURE_PREFIX}-gwy-$LOCATION --name gateway-vmss-$gitCommit --force-deletion
+  echo "Success step 7c ✅ - Remove RP VMSS and GWY VMSS"
+  echo -e "Success step 7 ✅ - Final preperation before deploying RP\n"
+}
+
+# Function to fully deploy 12 deployments in case they are needed
+fully_deploy_resources() {
+    # Don't skip deployment creation when SKIP_DEPLOYMENTS is set to "false" 
+    if [[ "${SKIP_DEPLOYMENTS}" == "false" ]]; then
+        echo "'SKIP_DEPLOYMENTS' env var is set to false. Don't skip full deployments of RP and GYW."
+        make go-verify deploy
+        return 0
+    fi
+
+    num_deployment=0
+    check_deployment ${AZURE_PREFIX}-global rp-global-${LOCATION} && num_deployment=$(( num_deployment + 1))
+    check_deployment ${AZURE_PREFIX}-subscription rp-production-subscription-${LOCATION} && num_deployment=$(( num_deployment + 1))
+
+    gitCommit=$(git rev-parse --short=7 HEAD)
```

Comment:
1. Double quotes
2. I see this git command being ran a lot, can we run it once in the functions referencing this, then pass it in as an argument? If the commit doesn't change, it should only be ran once

Reply:
Done